R/soar-Dashboard.R

```diff
@@ -63,105 +111,31 @@ ui <- dashboardPage(
+                textInput("grid_cell_size", "Grid Cell Size: "), textInput("MinCellDen", "Minimum Cell Density:"), 
+                actionButton("cus_down", "Down-Sample data with custom settings"))),
       tabPanel("Raw Data", DT::dataTableOutput("raw_data")),
-      tabPanel("Download table", 
-              # h2("Ability to download the table will be added here"),
-               radioButtons("tableCols", label = "Columns in Downloadable table",
+      tabPanel("Download table",
+               radioButtons("table_cols", label = "Columns in Downloadable table",
                                   choices = list("Minimal" = 1, "Default" = 2, "All Columns" = 3, "Custom" = 4), 
                                   selected = 1),
 #ADD - Download link for table
-               downloadButton('downloadData', label = "Download Table"),
-               conditionalPanel(condition = "input.tableCols == 1", 
-                                checkboxGroupInput("tableMin", label = h4("Minimal Options:"), choices = list(
-                                                    "Species" = 229, "Taxon ID" = 175, "Key" = 220, "Decimal Latitude" = 133, "Decimal Longitude" = 134)
-                                                     , 
+               downloadButton('download_data', label = "Download Table"),
+               conditionalPanel(condition = "input.table_cols == 1", 
+                                checkboxGroupInput("table_min", label = h4("Minimal Options:"), choices =
+                                                   min_choice(), 
                                                    selected = c(133, 134, 175, 220, 229))),
-               conditionalPanel(condition = "input.tableCols == 2",
-                                checkboxGroupInput("tableDef", label= h4("Default Options:"), choices = list("Refrences" = 43, "Rights" = 47,
-                                                   "Rights holder" = 48, "Institution Code" = 60, "Information Withheld" = 65,
-                                                   "Catalog Number" = 69, "Sex" = 75, "Life Stage" = 76, "Verbatim Event Date" = 106,
-                                                   "Country Code" = 121, "Decimal Latitude" = 133, "Decimal Longitude" = 134, 
-                                                   "Coordinate Unceartanty" = 135, "Taxon ID" = 175, "Scientific Name" = 183,
-                                                   "Kingdom" = 191, "Phylum" = 192, "Class" = 193, "Order" = 194, "Family" = 195, 
-                                                   "Genus" = 196, "Subgenus" = 197, "Specific Epithet" = 198, 
-                                                   "Infraspecific Epithet" = 199, "Taxon Rank" = 200, "Dataset Key" = 207, 
-                                                   "Has Geospatial Issues" = 219, "Species" = 229, "Generic Name" = 230
-                                                   ),
+               conditionalPanel(condition = "input.table_cols == 2",
```

Comment:
This looks better but it still appears vulnerable to the ordering of the column names since you are using numeric indices rather than strings. If GBIF decides to change the ordering of the columns the code will bug.

Reply:
This is already an issue as we are working off of a transcribed list in a .csv file. GBIF changing the order of their columns will not affect the .csv. If the rest of the code is dependent on the file then it probably wouldn't be a good idea to have a different part of the code depended on something else.